File: Examples/VisualBasic/ModifyingAndConvertingImages/PNG/CompressingFiles.vb
Imports Aspose.Imaging.ImageOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.PNG
    Class CompressingFiles
        Public Shared Sub Run()
            ' ExStart:CompressingFiles
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_PNG()

            ' Load an image from file (or stream)
            Using image__1 As Image = Image.Load(dataDir & Convert.ToString("aspose_logo.png"))
                ' Loop over possible CompressionLevel range
                For i As Integer = 0 To 9
                    ' Create an instance of PngOptions for each resultant PNG, Set CompressionLevel and  Save result on disk
                    Dim options As New PngOptions()
                    options.CompressionLevel = i
                    image__1.Save(i & "_out.png", options)
                Next
            End Using
            ' ExEnd:CompressingFiles
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/PNG/CompressingFiles.cs
// GIST-ID: de1271a2dd80bf8214dcb38bf27a91a7
using Aspose.Imaging.ImageOptions;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.PNG
{
    class CompressingFiles
    {
        public static void Run()
        {
            Console.WriteLine("Running example CompressingFiles");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_PNG();

            // Load an image from a file (or stream)
            using (Image image = Image.Load(dataDir + "aspose_logo.png"))
            {
                // Loop over possible CompressionLevel range
                for (int i = 0; i <= 9; i++)
                {
                    // Create an instance of PngOptions for each resultant PNG, set CompressionLevel, and save the result on disk
                    PngOptions options = new PngOptions();
                    options.CompressionLevel = i;
                    image.Save(i + "_out.png", options);
                }
            }

            Console.WriteLine("Finished example CompressingFiles");
        }
    }
}